packages/x-date-pickers/src/DigitalClock/DigitalClock.tsx

```diff
@@ -219,7 +220,8 @@ export const DigitalClock = React.forwardRef(function DigitalClock<TDate extends
       return;
     }
     const offsetTop = activeItem.offsetTop;
-    if (autoFocus || !!focusedView) {
+    if ((autoFocus || !!focusedView) && activeItem !== lastActiveRef.current) {
```

Comment:
Fix the issue of stealing the focus when closing a picker with DigitalClock:

https://github.com/mui/mui-x/assets/4941090/4eac0e51-56af-49b0-a331-df47887f8429